examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1175 @@
+
+        {
+          AssertThrow(cell != dof_handler.end(), ExcInternalError());
+          std::cout << " - Found point with real coordinates: " << p_real
+                    << std::endl;
+          std::cout << "   - in cell with vertices:";
+          for (const auto &v : cell->vertex_indices())
+            std::cout << " (" << cell->vertex(v) << ")";
+          std::cout << std::endl;
+          std::cout << "   - with coordinates on the unit cell: (" << p_unit
+                    << ")" << std::endl;
+        }
+
+        // Now that we have determined $K$ and $\hat{\boldsymbol{x}}_K$, we can
+        // perform the evaluation of the finite element solution at this
+        // point. In the following, we show three approaches for this
+        // purpose. In the first approach, we follow a traditional technique
+        // by using FEValues based on a cell-specific quadrature rule
+        // consisting of the unit point.
+        std::cout << " - Values at point:" << std::endl;
+
+        {
+          FEValues<dim> fe_values(mapping,
+                                  fe,
+                                  Quadrature<dim>(p_unit),
+                                  update_values);
+          fe_values.reinit(cell);
+
+          std::vector<double> quad_values(1);
+          fe_values.get_function_values(vector, quad_values);
+          const double value_0 = quad_values[0];
+          std::cout << "  - " << value_0 << " (w. FEValues)" << std::endl;
+          values_line.push_back(value_0);
+        }
+
+        // The second approach considers FEPointEvaluation, which directly
+        // takes a list of unit points for the subsequent evaluation.
+        {
+          std::vector<double> cell_vector(fe.n_dofs_per_cell());
+          cell->get_dof_values(vector, cell_vector.begin(), cell_vector.end());
+
+          FEPointEvaluation<1, dim> fe_point(mapping, fe, update_values);
+          fe_point.reinit(cell, ArrayView<const Point<dim>>(p_unit));
+          fe_point.evaluate(cell_vector, EvaluationFlags::values);
+          const auto value_1 = fe_point.get_value(0);
+          std::cout << "  - " << value_1 << " (w. FEPointEvaluation)"
+                    << std::endl;
+        }
+
+        // Finally, in the third approach, the function
+        // VectorTools::point_value() is considered. It performs both
+        // the search of the surrounding cell and the evaluation at the
+        // requested point. However, its application is limited
+        // to a serial run of the code.
+        {
+          const auto value_2 =
+            VectorTools::point_value(dof_handler, vector, p_real);
+          std::cout << "  - " << value_2 << " (w. VectorTools::point_value())"
+                    << std::endl;
+          std::cout << std::endl;
+        }
+      }
+
+    // We output the requested points together with the corresponding
+    // evaluated solution to a CSV file.
+    std::cout << " - writing csv file" << std::endl;
```

Comment:
Is the underlying goal here to illustrate that there many ways to achieve the same thing, or do you wish to demonstrate that some approaches are more efficient / less time consuming?
I am not sure I understand the full motivation for showing three different ways to achieve the same purpose. Generally, my understanding is that it is a bad thing when you have three ways of doing the same things, it's confusing for a new reader.